google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/Heartbeat.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.models;
+
+import com.google.api.core.InternalApi;
+import com.google.bigtable.v2.ReadChangeStreamResponse;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import java.io.Serializable;
+import javax.annotation.Nonnull;
+
+public final class Heartbeat implements ChangeStreamRecord, Serializable {
+  private static final long serialVersionUID = 7316215828353608504L;
+  private final com.google.protobuf.Timestamp lowWatermark;
+  private final ChangeStreamContinuationToken changeStreamContinuationToken;
+
+  private Heartbeat(
+      com.google.protobuf.Timestamp lowWatermark,
+      ChangeStreamContinuationToken changeStreamContinuationToken) {
+    this.lowWatermark = lowWatermark;
+    this.changeStreamContinuationToken = changeStreamContinuationToken;
+  }
+
+  @InternalApi("Used in Changestream beam pipeline.")
+  public ChangeStreamContinuationToken getChangeStreamContinuationToken() {
+    return changeStreamContinuationToken;
+  }
+
+  @InternalApi("Used in Changestream beam pipeline.")
+  public com.google.protobuf.Timestamp getLowWatermark() {
+    return lowWatermark;
+  }
+
+  /** Wraps the protobuf {@link ReadChangeStreamResponse.Heartbeat}. */
+  @InternalApi("Used in Changestream veneer client.")
+  public static Heartbeat fromProto(@Nonnull ReadChangeStreamResponse.Heartbeat heartbeat) {
+    return new Heartbeat(
+        heartbeat.getLowWatermark(),
+        ChangeStreamContinuationToken.fromProto(heartbeat.getContinuationToken()));
```

Comment:
Do you really need the ChangeStreamContinuationToken wrapper class? What's wrong with having fromProto(readChangeStreamResponse.Heartbeat heartbeat) {
return new Heartbeat(getLowWatermark(), getContinuationToken()
}?

Reply:
I'm not sure if it's a good idea to expose the raw proto(i.e. com.google.bigtable.v2.StreamContinuationToken) to the beam code, therefore I use the ChangeStreamContinuationToken as a wrapper for com.google.bigtable.v2.StreamContinuationToken. 

For example, the beam code might use a CloseStream to create a ReadChangeStreamQuery like:
```
ReadChangeStreamQuery.creat("tableId").streamPartition("", "").continuationTokens(closeStream.getChangeStreamContinuationTokens());
```

If we don't use the ChangeStreamContinuationTokens wrapper, we will need to return `List<com.google.bigtable.v2.StreamContinuationToken> getStreamContinuationTokens()` from CloseStream.

Reply:
Discussed offline. Let's keep it.